crates/bevy_app/src/app_builder.rs

```diff
@@ -292,7 +292,7 @@ impl AppBuilder {
     /// and inserting a `Events::<T>::update_system` system into `CoreStage::First`.
     pub fn add_event<T>(&mut self) -> &mut Self
     where
-        T: Component,
+        T: Resource,
```

Comment:
Yay separating out these bounds! I'll need to swap this to `T: Component + Resource` for #1626, but that shouldn't be any issue.